Blind manuscript submission changes (#1)
fchampalimaud/syringe.pump.manuscript: Manuscript/Chapters/Discussion.tex

Inspired by previous projects (\citep{Amarante2019, Wijnen2014}), we present a fully integrated, and characterized system for microliter range fluid delivery targeted towards neuroscience behavior experiments. 
 
We provide an alternative to gravity-based passive systems, by designing a scalable and affordable solution that allows for dynamic control over reward delivery, without suffering from the long-term stability issues often associated with the former systems. Additionally, we also highlight the afforded control gained over flow-rate that such an non-passive system allows.

> In addition to the mechanical design of the syringe pump, we also developed a custom PCB that not only controls the presented pump but can also serve as a generic controller for various types of stepper motor-based syringe pumps. This allows users to design and assemble their own pump configurations. Furthermore, the integration of Harp protocol at the firmware level, enables precise timestamping and synchronization with other Harp devices, facilitating scalable and reliable experimental data collection pipelines.

Reviewer: ```suggestion
In addition to the mechanical design of the syringe pump, we also developed a custom PCB that not only controls the presented pump but can also serve as a generic controller for various types of stepper motor-based syringe pumps. This allows users to design and assemble their own pump configurations. Furthermore, by implementing the Harp protocol, the firmware affords precise timestamping and synchronization with other Harp devices, facilitating scalable and reliable experimental data collection pipelines.
```

> All the necessary design and production files for the syringe pump, including the PCB, firmware, and GUI, have been made available, along with detailed assembly instruction to simplify the assembly process. By providing these files, our goal is not only to enable users to replicate the design but also to make modifications according to their specific needs.

Reviewer: ```suggestion
All the necessary design and production files for the system, including the PCB, firmware, and GUI, have been made freely available, along with detailed assembly instructions. By providing these files, our goal is not only to enable users to replicate the design but also to make modifications according to their specific needs.
```

> To ensure seamless integration with already-existing experimental rigs (such as replacing gravity-based fluid delivery systems), we offer users with several options to interface with the system, affording a very flexible configuration and compatibility. By using pre-existent controllers available in the experimental setups, the syringe pump can be fully controlled through its low level interface, a trigger based protocol, the GUI or the Harp API. The integration of this API is already available in Bonsai, which is particularly relevant since it leverages the capabilities of the syringe pump, by offering a powerful tool for real-time data stream processing, video acquisition, close-loop tasks and real time data visualization, in a visual programming language environment. 

Reviewer: ```suggestion
To ensure seamless integration with already-existing experimental rigs (such as replacing gravity-based fluid delivery systems), our design provides different levels of operation control. Specifically, the syringe pump can be fully controlled through its low-level interface, trigger-baed protocol, graphical user interface or, more generically, via the Harp API. The integration of this API is already available in the Bonsai programming language. This integration is particularly relevant since it offers a powerful ecosystem of tools for real-time data stream processing, video acquisition, close-loop tasks and real-time data visualization, in a visual programming language environment. 
```

> We also designed a novel and simple protocol which allowed us provide an extensive characterization of trial-to-trial variability in the fluid delivery dynamics. An import stack of tests when considering the intended use of this system for rodent behavior experiments. Traditionally, fluid volume characterization relies on repeating multiple fluid deliveries and then weighing the total volume. However, this approach only provides the mean volume over multiple deliveries and fails to assess variability across single fluid delivery events. Due to the mechanical nature of the syringe pump design, factors such as defective rods, material hysteresis, syringes construction, among others, can potentially introduce variations in small-volume deliveries. This makes it important to assess single delivery events. However, we acknowledge that measuring single small-volume delivery events presents a technical challenge in that equipment capable of reliably measuring such small amounts is rarely available.

Reviewer: ```suggestion
We also report on the fluid-delivery capabilities of the system by providing a comprehensive stack of tests and benchmarks. Traditionally, fluid volume characterization relies on repeating multiple fluid deliveries and then weighing the total volume. However, this approach only provides the mean volume over multiple deliveries and fails to assess variability across single fluid delivery events. Due to the mechanical nature of the syringe pump design, factors such as defective rods, material hysteresis, syringe construction, among others, can easily introduce variations in microliter-range protocols. 
```

> Importantly, we also tested the compatibility of the system with common neuroscience experimental demands by deploying it during rodent behavior, where we successfully demonstrated the controlled variation of delivered reward amounts. Additionally, we conducted electrophysiological recording experiments, confirming that no electrical artifacts were observed.

Reviewer: ```suggestion
Finally, we also tested the compatibility of the system with common neuroscience experimental demands by successfully deploying it in rodent behavior experiments and, highlighting the system's compatibility electrophysiological recordings.```
